### PR TITLE
fix(sdk): fix attach in offline mode

### PR DESCRIPTION
**Pull request by kptkin**
Fixes WB-NNNN
Fixes #NNNN

Description
-----------
What does the PR do?

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/interface/interface_sock.py [hunk 1]
```diff
@@ -48,6 +48,8 @@ def _assign(self, record: Any) -> None:
 
     def _publish(self, record: "pb.Record", local: Optional[bool] = None) -> None:
         self._assign(record)
+        print("InterfaceSock._publish", record)
+        print("=" * 80)
```
Comment (kptkin) [inline wandb/sdk/interface/interface_sock.py:52]:
```suggestion
```

File: wandb/sdk/wandb_init.py [hunk 2]
```diff
@@ -873,18 +873,12 @@ def _attach(
...
     settings: Settings = copy.copy(_wl._settings)
-    settings.update(
-        {
-            "run_id": attach_id,
-            "_start_time": response["_start_time"],
-            "_start_datetime": response["_start_datetime"],
-        },
-        source=Source.INIT,
-    )
+    settings.update(response, source=Source.INIT)
```
Comment (kptkin) [inline wandb/sdk/wandb_init.py:881]: is this safe?

**Comment by raubitsj** [comment]
follow up to:
https://github.com/wandb/wandb/pull/5144

**Comment by raubitsj** [comment]
chatted with @kptkin,  going to:
- unify deliver_run and publish_run
- move attach handling from sender to handler
- unify offling_run and online_run in the run class

**Pull request closed without merging**